pkg/client/interfaces.go

```diff
@@ -77,26 +79,41 @@ type RetrieveConfig struct {
 
 // SonobuoyClient is a high-level interface to Sonobuoy operations.
 type SonobuoyClient struct {
-	RestConfig *rest.Config
-	Client     kubernetes.Interface
-	// TODO (timothysc) cache dynamic client
+	RestConfig    *rest.Config
+	client        kubernetes.Interface
+	dynamicClient dynamic.ClientPool
 }
 
 // NewSonobuoyClient creates a new SonobuoyClient
 func NewSonobuoyClient(restConfig *rest.Config) (*SonobuoyClient, error) {
 	sc := &SonobuoyClient{
-		RestConfig: restConfig,
+		RestConfig:    restConfig,
+		client:        nil,
```

Comment:
it always defaults to nil, but meh I don't care.

Reply:
leaving uninitialised fields makes me nervous 😓